### PR TITLE
Blog Post about action composition with Mutiny

### PR DESCRIPTION
**Pull request by cescoffier**
A blog post illustrating action composition with Mutiny.

### CODE REVIEW

File: _posts/2020-09-02-mutiny-redis.adoc [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+So, we want to transform our key into a Uni (*transformToUni*).
+
+When you have a stream of items and need to invoke an asynchronous action for each item, you must choose how you will merge the results. Mutiny provides two strategies:
+
+- *merge -* as soon as the item produced by the `Uni` is received we send it downstream
+- *concatenate* - we preserve the order of the input stream to be sure that the items are sent downstream in the same order
+
+Let's illustrate this. Imagine we have the keys `1`, `2`, `3` and to the stream `{1, 2, 3}`. Also, let's consider that in our Redis database, the key `1` is associated to `A`, `2` to `B` and `3` to `C`. 
+
+If you use the *merge* strategy, we are retrieving the associated objects we the order is undetermined. We can end up with `{A, C, B}` or `{B, A, C}`. It depends on many factors, such as the latency,  scheduling, load and so on. However, it also means we can retrieve all the associated objects concurrently and produce the resulting stream without taking care of the order.
```
Comment (kenfinnigan) [inline _posts/2020-09-02-mutiny-redis.adoc:89]: "objects we the order" doesn't make sense to me

Comment (kenfinnigan) [reply _posts/2020-09-02-mutiny-redis.adoc:89]: Is it meant to be "where"?

Comment (kenfinnigan) [reply _posts/2020-09-02-mutiny-redis.adoc:89]: Remove the extra space after "latency,"

File: _posts/2020-09-02-mutiny-redis.adoc [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+If you use the *concatenate* strategy, it preserves the order from the input stream. So, it will always produce `{A, B, C}`. While it may be desirable, it may reduce the ability to retrieve the object concurrently, as Mutiny has to *wait* for all the retrieval of all the previous objects. For example, if Mutiny receives `C` first, it needs to wait for `A` and `B` before sending `C` downstream. 
+
+In our context, let's not preserve the order and use the *merge* strategy. So we use `transformToUniAndMerge`. 
+
+If you run the code multiple times, you might see order changes in the resulting array.
+
+Ok, step 2 done. Let's focus on the final steps: accumulate the objects into a `JsonArray`, and produce a `Uni<JsonArray>`, containing all the objects. Mutiny provides methods to gather items from a stream into lists, maps, sets, but not `JsonArray`. Fortunately, it offers a method that let you do this:
```
Comment (kenfinnigan) [inline _posts/2020-09-02-mutiny-redis.adoc:97]: What's the "it" that offers a method that allows you to do it, as you just mentioned Mutiny doesn't provide a method

File: _posts/2020-09-02-mutiny-redis.adoc (hunk 1 above)
Comment (cescoffier) [reply _posts/2020-09-02-mutiny-redis.adoc]: It meant to be: "If you use the *merge* strategy, we are retrieving the associated objects in an undetermined order."

File: _posts/2020-09-02-mutiny-redis.adoc (hunk 2 above)
Comment (cescoffier) [reply _posts/2020-09-02-mutiny-redis.adoc]: Rephrased.

**Pull request merged**